src/stemmabench/algorithms/stemma.py

```diff
@@ -0,0 +1,209 @@
+          With manuscript as labels as keys and reference of manuscripte as value.
+        - fitted (bool): Indicates if the stemma has been built.
+        - generation_info (dict): Dictionary containing info about how the tree was generated.
+    """
+
+    def __init__(self,
+                 folder_path: Union[str, None] = None,
+                 edge_file: Union[str, None] = None,
+                 generation_info: Dict[str, Any] = {}) -> None:
+        """A class to perform variant generation.
+        To instansite the class use one of the build methods.
+
+        ### Args:
+            - folder_path (str, Optional): The path to the folder that contains the texts.
+            - edge_file (str, Optional): An edge file from which the tree can be built. 
+            !!! The labels used in the edge file must be the same as the name used for the text .txt names !!!
+            - generation_info (dict, Optional): A dictionnary containing information about the stemma's generation.
+
+        ### Generation_info:
+            This dictionary is ment to contain all manner of miscellaneous information relating to the generation of the stemma.
+            The following section is a suggestion of the possible contents of the dictionary.
+        #### Content:
+            - config (str, Optional): The path to the config file used to generate the tree.
+            - has_ground_truth (bool, Optional): Boolean indicating if the true tree is known. If false the path_to_original attribute will indicate the 
+            estimated original text.
+        """
+        self._set_folder_path(folder_path=folder_path)
+        self._edge_file: Union[str, None] = edge_file
+        self._generation_info: Union[Dict[str, Any], None] = generation_info
+        self._root: Union[ManuscriptBase, None] = None
+        self._text_lookup: Dict[str, ManuscriptBase] = {}
+        self._fitted: bool = False
+
+    @property
+    def root(self):
+        return self._root
+
+    @property
+    def folder_path(self):
+        return self._folder_path
+    
+    @property
+    def edge_file(self):
+        return self._edge_file
+    
+    @property
+    def fitted(self):
+        return self._fitted
+    
+    @property
+    def text_lookup(self):
+        return self._text_lookup
+    
+    @property
+    def generation_info(self):
+        return self._generation_info
+    
+    def _set_folder_path(self, folder_path: Union[str, None]) -> None:
+        """Checks that the folder path is an existing directory and sets the folder_path attribute.
+        
+        ### Args:
+            - folder_path (str): The path to the folder containing all the texts.
+
+        ### Raises:
+            - ValueError: If the specified folder_path is not an existing directory.
+        """
+        if folder_path and not os.path.isdir(folder_path):
+            raise ValueError("The folder_path specified is not an existing folder path.")
```

Comment:
You can add f-string in the error message for clarity:

`ValueError("The folder path {folder_path} specified is not an existing folder path.")`

Reply:
Added f-strings